### PR TITLE
Create subject access report endpoint and protect endpoints with HMPPS OAuth

### PR DESCRIPTION
**Pull request by CslBing**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppssubjectaccessrequestapi/controllers/SubjectAccessRequestController.kt
Comment (HorstmannA) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppssubjectaccessrequestapi/controllers/SubjectAccessRequestController.kt:1]: Small thing - 'creatSubjectAccessReport' should be 'createSubjectAccessReport'

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppssubjectaccessrequestapi/controllers/SubjectAccessRequestController.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@RequestMapping("/api/")
+class SubjectAccessRequestController {
+
+  @PostMapping("creatSubjectAccessReport")
```
Comment (HorstmannA) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppssubjectaccessrequestapi/controllers/SubjectAccessRequestController.kt:11]: Small thing - 'creatSubjectAccessReport' should be 'createSubjectAccessReport'

**Review by AprilMD — DISMISSED** [review]

**Review by AprilMD — DISMISSED** [review]

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppssubjectaccessrequestapi/integration/AuthTest.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      .exchange()
+      .expectStatus()
+      .isForbidden
+      .expectBody()
+  }
+
+  @Test
+  fun `Providing malformed JWT to a secured endpoint returns 403`() {
+    webTestClient.get()
+      .uri("/api/creatSubjectAccessReport")
```
Comment (HorstmannA) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppssubjectaccessrequestapi/integration/AuthTest.kt:18]: Small typo

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppssubjectaccessrequestapi/integration/AuthTest.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package uk.gov.justice.digital.hmpps.hmppssubjectaccessrequestapi.integration
+
+import org.junit.jupiter.api.Test
+class AuthTest : IntegrationTestBase() {
+  @Test
+  fun `Providing no JWT to a secured endpoint returns 403`() {
+    webTestClient.get()
+      .uri("/api/creatSubjectAccessReport")
```
Comment (HorstmannA) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppssubjectaccessrequestapi/integration/AuthTest.kt:8]: Small typo

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppssubjectaccessrequestapi/integration/AuthTest.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      .expectStatus()
+      .isForbidden
+  }
+
+  @Test
+  fun `Providing JWT signed by wrong private key to a secured endpoint returns 403`() {
+    val jwt = "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJhcHByb3ZlZC1wcmVtaXNlcy1hcGkiLCJncmFudF90eXBlIjoiY2xpZW50X2NyZWRlbnRpYWxzIiwic2NvcGUiOlsicmVhZCJdLCJhdXRoX3NvdXJjZSI6Im5vbmUiLCJpc3MiOiJodHRwOi8vbG9jYWxob3N0OjkwOTEvYXV0aC9pc3N1ZXIiLCJleHAiOjI2NTk3MDQ5NDAsImF1dGhvcml0aWVzIjpbIlJPTEVfSU5URVJWRU5USU9OUyIsIlJPTEVfT0FTWVNfUkVBRF9PTkxZIiwiUk9MRV9DT01NVU5JVFkiLCJST0xFX0dMT0JBTF9TRUFSQ0giLCJST0xFX0NPTU1VTklUWV9VU0VSUyIsIlJPTEVfUklTS19TVU1NQVJZIl0sImp0aSI6ImlTSEtsTXJ1aXFUNjF0dTNXVFFqckE2WWJfTSIsImNsaWVudF9pZCI6ImFwcHJvdmVkLXByZW1pc2VzLWFwaSJ9.Cr7Nl09vjUpyieddsJwyQF02nmqhR6PbM4xePA47ukkyhhctE4SwqpOAO5D5OIstr9ePnlmF_Tug7HZ6-SLF8lBnN9C_M2-74d8127gPkQxjWsGnAKIxAGDnwLjtwV1UpSvS0p-Phg3cBTGiq6_HABEuh2JSD67eJS0ZaqNPUXXp2kTfi1ZJXA1ysxFKvAP5qYHbBpYWfvFq9Wkpsrq4sM41yjzS7hmkpaEUAYvKUdYefeRAT6nMCU6pfkEOoCmXkMTf6n6rJ1HxxTvkucZwEQk1dOKZUH0d_AOjZy5RAXiSRzgiYsMfB02gvn2T0FfOyjkjKXgVDsFc2yf3bd6P0g"
+
+    webTestClient.get()
+      .uri("/api/creatSubjectAccessReport")
```
Comment (HorstmannA) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppssubjectaccessrequestapi/integration/AuthTest.kt:30]: Small typo

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppssubjectaccessrequestapi/integration/AuthTest.kt (hunk 3 above)
Comment (CslBing) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppssubjectaccessrequestapi/integration/AuthTest.kt]: Fixed typo

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppssubjectaccessrequestapi/integration/AuthTest.kt (hunk 4 above)
Comment (CslBing) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppssubjectaccessrequestapi/integration/AuthTest.kt]: Fixed typo

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppssubjectaccessrequestapi/integration/AuthTest.kt (hunk 2 above)
Comment (CslBing) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppssubjectaccessrequestapi/integration/AuthTest.kt]: Fixed typo

**Review by HorstmannA — APPROVED** [review]

**Pull request merged**